### PR TITLE
Fix/group row hoisting

### PR DESCRIPTION
**Pull request by kariyngva**
# Description

When nesting `row` fields withing a `group` field, the fields within the row get hoisted out of the group field in the GraphQL schema, making the where input type incorrect. This does not seem to be an issue in the REST API.

Given a collection with the following fields:

```ts
    {
      type: 'group',
      name: 'testGroup',
      fields: [
        {
          type: 'row',
          fields: [
            {
              type: 'text',
              name: 'textInRowInGroup',
              defaultValue: 'textInRowInGroup',
            },
          ],
        },
        {
          type: 'text',
          name: 'textInGroup',
          defaultValue: 'textInGroup',
        },
      ],
    },
```

According to the GraphQL schema `textInRowInGroup` is in the root of the collection, not `testGroup__textInRowInGroup`. Querying after it with a `where` query will lead to an error.

```gql
{
  Posts(where: { textInRowInGroup: { equals: "textInRowInGroup" } }) {
    totalDocs
  }
}
```

```json
{
  "errors": [
    {
      "message": "The following path cannot be queried: textInRowInGroup",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": ["Posts"],
      "extensions": {
        "name": "QueryError",
        "data": [
          {
            "path": "textInRowInGroup"
          }
        ]
      }
    }
  ]
}
```

Querying after the correct field name also returns an error:

```gql
{
  Posts(
    where: { testGroup__textInRowInGroup: { equals: "textInRowInGroup" } }
  ) {
    totalDocs
  }
}
```

```json
{
  "errors": [
    {
      "message": "Field \"testGroup__textInRowInGroup\" is not defined by type \"Post_where\". Did you mean \"testGroup__textInGroup\" or \"textInRowInGroup\"?",
      "locations": [
        {
          "line": 2,
          "column": 17
        }
      ]
    }
  ]
}
```

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes

### CODE REVIEW

**Review by JarrodMFlesch — CHANGES_REQUESTED** [review]

File: test/_community/collections/Posts/index.ts [hunk 1]
```diff
@@ -10,6 +10,37 @@ export const PostsCollection: CollectionConfig = {
       name: 'text',
       type: 'text',
     },
+    {
```
[the remark this answers is not in the document]
Comment (kariyngva) [reply test/_community/collections/Posts/index.ts:13]: Cool, I'll move it over. You wanna change the labeling at all? I feel `should query testGroup__textInRowInGroup` isn't really in line with the others does`field is queriable` make more sense?

File: src/graphql/schema/recursivelyBuildNestedPaths.ts [hunk 2]
```diff
@@ -13,8 +13,16 @@ const recursivelyBuildNestedPaths = (parentName: string, field: FieldWithSubFiel
       const nestedFieldName = fieldAffectsData(nestedField) ? `${field.name}__${nestedField.name}` : undefined;
 
       if (getFieldSchema) {
+        let maybeTransformedNestedField = nestedField;
+        if (nestedField.type === 'row') {
```
[the remark this answers is not in the document]
Comment (kariyngva) [reply src/graphql/schema/recursivelyBuildNestedPaths.ts:17]: Yeah I'll check that out too

File: test/_community/collections/Posts/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JarrodMFlesch) [reply test/_community/collections/Posts/index.ts:13]: Yeah maybe, `Can query Group > Row > Field`

File: src/graphql/schema/recursivelyBuildNestedPaths.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kariyngva) [reply src/graphql/schema/recursivelyBuildNestedPaths.ts]: Verified collapsible's had the same issue, should be fixed now.

File: test/_community/collections/Posts/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JarrodMFlesch) [reply test/_community/collections/Posts/index.ts:13]: Yep! Once you clean up the _community folder I feel good merging it in 👍

[the remark this answers is not in the document]
Comment (kariyngva) [reply test/_community/collections/Posts/index.ts]: All clean, we should be good to go 🥳

**Comment by JarrodMFlesch** [comment]
@kariyngva I have some changes that I want to push back to your fork, but for some reason I am unable to 🤔

**Comment by kariyngva** [comment]
> @kariyngva I have some changes that I want to push back to your fork, but for some reason I am unable to 🤔
> 
> I think there is a checkbox on this PR that you can check and I can push my changes up to your forked branch. Can you check and let me know?

If you mean the "Allow edits by maintainers" it's checked. I added you as a collaborator on my repo so you should be able to push once you accept the invitation.

**Comment by JarrodMFlesch** [comment]
@kariyngva oh man, I think it has to do with me merging master into this - which had an update to the .github/worflows/test.yml file and it will not allow me to push that up to your repo. Sorry for the hassle, I wonder if you can merge with master and the I can pull that into my branch....

**Comment by kariyngva** [comment]
> @kariyngva oh man, I think it has to do with me merging master into this - which had an update to the .github/workflows/test.yml file and it will not allow me to push that up to your repo. Sorry for the hassle, I wonder if you can merge with master and the I can pull that into my branch....

No worries, updated my fork and merged master into this branch

**Comment by edtorba** [comment]
Hey guys, I am afraid it's still not okay by the looks...

Here's the setup I got:

```js
const Mycollection: CollectionConfig = {
  slug: 'mycollection',
  access: {
    create: () => true,
    read: () => true,
    update: () => true,
    delete: () => true,
  },
  fields: [
    {
      label: 'My Thing',
      description: 'Description for my thing...',
      fields: [
        {
          label: 'My Group',
          name: 'mygroup',
          type: 'group',
          fields: [
            {
              name: 'foo',
              type: 'number',
              required: true,
              defaultValue: 1,
            },
            {
              type: 'row',
              fields: [
                {
                  name: 'bar',
                  type: 'number',
                  defaultValue: 2,
                },
                {
                  name: 'tar',
                  type: 'number',
                  required: true,
                  defaultValue: 3,
                },
              ]
            },
          ],
        },
      ],
    },
  ],
};
```
The `mygroup__foo` if fully omitted and throws an error:

```
"message": "Field \"mygroup__foo\" is not defined by type \"Mycollection_where\". Did you mean \"mygroup__bar\"?",
```

Note: without the nested `row`, all is working as expected.

I can spin up a separate issue with the failing setup if needed 👍

**Comment by JarrodMFlesch** [comment]
Hey @edtorba if you could open a new one that would be sweet! 

I think the error comes from this file in the reducer where we return the function recursively without spreading the currently built up paths along side it! https://github.com/payloadcms/payload/blob/master/src/graphql/schema/recursivelyBuildNestedPaths.ts

**Comment by JarrodMFlesch** [comment]
@edtorba I have a fix for this in this [PR](https://github.com/payloadcms/payload/pull/2769). Thanks for reporting 👍

**Pull request merged**